### PR TITLE
Add a formatted representation of the UpdatePageAction dataclass

### PR DESCRIPTION
**Pull request by mthaddon**
Applicable spec: N/A

### Overview

Add a formatted representation of the UpdatePageAction dataclass.

### Rationale

This relates to https://github.com/canonical/discourse-gatekeeper/issues/226. I've started by only adding for one dataclass to get some feedback on the approach before I apply it more widely.

### Module Changes

N/A

### Checklist

- [x] The [contributing guide](https://github.com/canonical/is-charms-contributing-guide) was applied
- [x] The documentation is generated using `src-docs`
- [x] The PR is tagged with appropriate label (`urgent`, `trivial`, `complex`)
- [x] Changelog has been updated
- [ ] Version has been incremented

Haven't incremented the version as I'm not sure if that's appropriate.

### CODE REVIEW

**Review by jdkandersson — APPROVED** [review]
Looks fine, please update the changelog. A lot of the issues with the checks should be resolved in https://github.com/canonical/discourse-gatekeeper/pull/231

**Review by cbartz — APPROVED** [review]

**Review by jdkandersson — APPROVED** [review]

**Review by arturo-seijas — DISMISSED** [review]

File: src/types_.py [hunk 1]
```diff
@@ -390,7 +390,11 @@ class UpdatePageAction(_UpdateActionBase):
     content_change: ContentChange
 
     def __str__(self):
```
[the remark this answers is not in the document]
Comment (mthaddon) [reply src/types_.py]: Good spot, thx.

**Comment by jdkandersson** [comment]
I think main can be merged in now which should address some of the failing checks

**Comment by mthaddon** [comment]
> I think main can be merged in now which should address some of the failing checks

```
$ git merge main
Already up to date.
```
I'll rerun the checks.

**Comment by jdkandersson** [comment]
The issues are:

* Black formatting doesn't seem to be applied
* The integration tests look for the name of the action in the log messages which used to be `UpdateAction(...)` or similar. Perhaps in the string representation include the action being taken? You can check the integration test for more details, please let me know if you need more help. It is case sensitive at the moment

**Review by arturo-seijas — APPROVED** [review]

**Review by Thanhphan1147 — APPROVED** [review]

**Review by jdkandersson — APPROVED** [review]

**Pull request merged**